builder/repo_control.go

```diff
@@ -224,6 +224,34 @@ func (b *Builder) SetExcludesRepo(reponame, pkgs string) error {
 	return DNFConf.SaveTo(b.Config.Builder.DNFConf)
 }
 
+// OverrideDnfConf creates a temporary dnf conf file based on the original
+// and overrides repo/URL settings.
+func (b *Builder) OverrideDnfConf(repoURLs map[string]string) (*os.File, error) {
+	// Create tmp dnf config based on original
+	tmpConf, err := ioutil.TempFile(os.TempDir(), "mixerConf")
+	if err != nil {
+		return tmpConf, err
+	}
+	origConf, err := ioutil.ReadFile(b.Config.Builder.DNFConf)
+	if err != nil {
+		return tmpConf, err
```

Comment:
i don't like the fact that the caller is responsible for closing tmpConf if the function fails here. This can lead to potential leak if this function is ever reused by someone that is not aware of this requirement.

Reply:
I'll move this to another function.